From: Jeff Moyer <jmoyer@redhat.com>
Date: Tue, 8 Dec 2009 20:59:16 -0500
Subject: [scsi] Correctly handle thin provisioning write error
Message-id: <1260305958-3673-6-git-send-email-jmoyer@redhat.com>
Patchwork-id: 21814
O-Subject: [RHEL6 PATCH 5/7] scsi: Correctly handle thin provisioning write error
Bugzilla: 528046
RH-Acked-by: David Milburn <dmilburn@redhat.com>

fixes bug 528046

commit 1304af2ff834452607163df88ede920504293b82
Author: Martin K. Petersen <martin.petersen@oracle.com>
Date:   Thu Nov 26 11:37:20 2009 -0500

    scsi: Correctly handle thin provisioning write error

    A thin provisioned device may temporarily be out of sufficient
    allocation units to fulfill a write request.  In that case it will
    return a space allocation in progress error.  Wait a bit and retry the
    write.

Signed-off-by: Jeff Moyer <jmoyer@redhat.com>
Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>

diff --git a/drivers/scsi/scsi_lib.c b/drivers/scsi/scsi_lib.c
index 5987da8..dd12183 100644
--- a/drivers/scsi/scsi_lib.c
+++ b/drivers/scsi/scsi_lib.c
@@ -859,6 +859,7 @@ void scsi_io_completion(struct scsi_cmnd *cmd, unsigned int good_bytes)
 				case 0x07: /* operation in progress */
 				case 0x08: /* Long write in progress */
 				case 0x09: /* self test in progress */
+				case 0x14: /* space allocation in progress */
 					action = ACTION_DELAYED_RETRY;
 					break;
 				default:
